### PR TITLE
feat(FX-3488): Notify user that they are opting back into email

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Feature**

This PR resolves [FX-3488]

⚠️ Changes behind a feature flag

### Description
* Modal has the same appearance as the "enable push notifications" modal
* Modal
  * title: Artsy would like to send you email notifications
  * description: After clicking “Save Alert”, you are opting in to receive alert notifications via email. You can update your email preferences by clicking into any alert listed in your profile tab and clicking “Update email preferences” underneath the “Email Alerts” toggle.
  * buttons: `Cancel`, `Accept`
* `Cancel` does not result in any API requests
* If a user clicked the `Accept` button, the modal will not be displayed next time when the “Email Alerts” toggle changes again
* Submitting the form (not closing the modal, but saving the alert) will trigger a mutation that updates a user's email preferences to `alerts_only`

#### Create alert flow
https://user-images.githubusercontent.com/3513494/141991203-175d158c-093e-4c28-9089-51e649b755e9.mp4

#### Edit alert flow
https://user-images.githubusercontent.com/3513494/141991257-2c0a041b-97d7-42b0-8a3a-9c0c1b199281.mp4

#### Should not display the modal if the "Email Alerts" toggle off and then back on
https://user-images.githubusercontent.com/3513494/142416610-7dafc4a1-3b0b-4c5d-945a-dc3038dfa494.mp4

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [x] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Notify user that they are opting back into email - dimatretyak

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[FX-3488]: https://artsyproduct.atlassian.net/browse/FX-3488?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/lib/Scenes/SavedSearchAlert/SavedSearchAlertForm.tsx [hunk 1]
```diff
@@ -66,6 +70,15 @@ export const SavedSearchAlertForm: React.FC<SavedSearchAlertFormProps> = (props)
...
         let result: SavedSearchAlertMutationResult
 
+        /**
+         * We perform the mutation only if
+         *  - the "Email Alerts" toggle was initially disabled and the user turned it on
+         *  - the user previously opted out of all marketing emails
+         */
+        if (enableSavedSearchToggles && !userAllowsEmails && !initialValues.email && values.email) {
+          await updateEmailFrequency("alerts_only")
+        }
```
Comment (dimatretyak) [inline src/lib/Scenes/SavedSearchAlert/SavedSearchAlertForm.tsx:80]: **question for reviewers**: should we call `updateEmailFrequency` when the "save alert" button is pressed, but the user previously opted out of emails and the "Email Alerts" toggle is enabled (for example, the user changed only name for saved search alert)?

Comment (mdole) [reply src/lib/Scenes/SavedSearchAlert/SavedSearchAlertForm.tsx:80]: good q @dimatretyak. I think it's the right call to only call the mutation if they have switched the toggle from "off" to "on" - we need to be sure they've seen the "you're opting back in" modal before they start receiving emails again.

I'm going to write a ticket + do a Gravity PR that will switch all `UserSearchCriteria` to `email: false` when a user opts out of marketing emails. That way, when a user opts out of all emails, then arrives on an "edit alert" screen, the toggle should already be switched to off, and they have to explicitly toggle back on, accept the modal, and save the alert before they start receiving alerts again.

**Review by mdole — APPROVED** [review]
whew! nice work here @dimatretyak and thanks for your patience with the copy questions and product decisions. This is looking good to me!

File: src/lib/Scenes/SavedSearchAlert/SavedSearchAlertForm.tsx (hunk 1 above)
Comment (dimatretyak) [reply src/lib/Scenes/SavedSearchAlert/SavedSearchAlertForm.tsx:80]: > That way, when a user opts out of all emails, then arrives on an "edit alert" screen, the toggle should already be switched to off, and they have to explicitly toggle back on, accept the modal, and save the alert before they start receiving alerts again.

Nice. I like this idea!

Comment (mdole) [reply src/lib/Scenes/SavedSearchAlert/SavedSearchAlertForm.tsx:80]: Ticketed here: https://artsyproduct.atlassian.net/browse/FX-3597

**Pull request merged**